linuxPackages.apfs: 0.3.14 -> 0.3.15

Luflosi opened the pull request:
https://github.com/linux-apfs/linux-apfs-rw/releases/tag/v0.3.15
Fixes #442599.

Also add an update script and remove the kernel version from the version for the following reasons:
- I don't see how this is useful since the information is already included in the kernel package itself
- We also don't include the version of the Python interpreter in the version of Python packages
- The normal update scripts fail to update this package, requiring a custom script
- Repology gets confused: https://repology.org/project/linux-apfs-rw/versions
- Example code in the NixOS wiki no longer includes the kernel version: https://wiki.nixos.org/w/index.php?title=Linux_kernel&oldid=17621#Packaging_out-of-tree_kernel_modules
- Several other kernel packages in Nixpkgs don't do this

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Luflosi committed:
linuxPackages.apfs: 0.3.14 -> 0.3.15

https://github.com/linux-apfs/linux-apfs-rw/releases/tag/v0.3.15

JohnRTitor commented:
> I don't see how this is useful since the information is already included in the kernel package itself

This is useful for knowing which kernel version this module is built for, from the outpath.

Ma27 commented on pkgs/os-specific/linux/apfs/default.nix:
```diff
@@ -5,20 +5,18 @@
+stdenv.mkDerivation (finalAttrs: {
   pname = "apfs";
-  version = "${tag}-${kernel.version}";
+  version = "0.3.15";
```
Please don't do this on a case-by-case basis and with the motivation being to appease other tools failing to handle this.

Luflosi committed:
linuxPackages.apfs: add passthru.updateScript

Add a custom update script because normal update scripts can't handle the special version, which includes the kernel version.
The simple solution of not including the kernel version in the package version was not accepted for some reason.

Luflosi commented:
I did not yet really understand the reason for keeping the kernel version as part of the package version but I bit the bullet and implemented my first custom update script.
